src/input_formatter/text_formatter.ts

```diff
@@ -0,0 +1,45 @@
+import { expressionFormatter } from "./expression_formatter";
+
+const inputSeparator = /([.!?]|\n|\$[^$]*\$)+/g;
+const wordSeparator = /[ ,]+/;
+
+/**
+ * this functions is the main function to format the full input string into a javastring array including formatted exrpessions
```

Comment:
What is the difference between a javastring array and a string array? Sorry, if this is common knowledge.